### PR TITLE
Add objectives support

### PR DESCRIPTION
**Pull request by int00h5525**

### CODE REVIEW

File: devel/moth.py [hunk 1]
```diff
@@ -59,6 +59,34 @@ def __init__(self, stream, name, visible=True):
         self.name = name
         self.visible = visible
 
+class PuzzleSuccess(dict):
```
Comment (nealey) [inline devel/moth.py]: Why not use a namedTuple?

File: devel/moth.py [hunk 2]
```diff
@@ -128,6 +163,14 @@ def read_stream(self, stream):
                     # Make sure this shows up in the header block of the HTML output.
                     self.files[val] = PuzzleFile(stream, val, visible=False)
                     self.scripts.append(val)
+                elif key == "objective":
+                    self.objective = val
+                elif key == "success":
+                    self.success = val
```
Comment (nealey) [inline devel/moth.py]: I think for the rfc822 parser we're going to need to read success.acceptable and success.mastery separately, right?

File: devel/moth.py (hunk 1 above)
Comment (int00h5525) [reply devel/moth.py]: namedtuple doesn't cleanly convert to a form that can be JSON-serialized, but I still wanted a way to enforce a schema

File: devel/moth.py (hunk 2 above)
Comment (int00h5525) [reply devel/moth.py]: Added in 03d1182e73ffc740d8ac4decdf89c8194d445cf4

**Pull request merged**